pkg/cli/debug/debug.go

```diff
@@ -595,8 +595,19 @@ func (o *DebugOptions) RunDebug() error {
 			// switch to logging output
 		case err == krun.ErrPodCompleted, err == conditions.ErrContainerTerminated:
 			resultPod, ok := containerRunningEvent.Object.(*corev1.Pod)
-			if ok && resultPod.Status.Reason == "NodeAffinity" && len(resultPod.Spec.NodeSelector) != 0 {
-				return fmt.Errorf("debug pod could not be scheduled: %v. To fix this you may want to create a new namespace with empty node selector and run the debug there. For example: oc adm new-project --node-selector=\"\" debug", resultPod.Status.Message)
+			if ok {
+				if resultPod.Status.Reason == "NodeAffinity" && len(resultPod.Spec.NodeSelector) != 0 {
+					return fmt.Errorf("debug pod could not be scheduled: %v. To fix this you may want to create a new namespace with empty node selector and run the debug there. For example: oc adm new-project --node-selector=\"\" debug", resultPod.Status.Message)
+				}
+				for _, c := range resultPod.Status.Conditions {
+					if c.Type == corev1.AlphaNoCompatGuaranteeDisruptionTarget {
```

Comment:
This is a very specific conditional. What does `AlphaNoCompatGuaranteeDisruptionTarget` mean?, why we can not gather logs when this exists.